### PR TITLE
Deprecate FilterOrderProvider

### PR DESCRIPTION
**Pull request by yawkat**
For #8422

### CODE REVIEW

File: http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java [hunk 1]
```diff
@@ -23,7 +23,9 @@
  *
  * @author James Kleeh
  * @since 1.0
+ * @deprecated Deprecated without replacement. This class is unused and will be removed in 4.0.0.
  */
+@Deprecated
```
Comment (sdelamo) [inline http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java:28]:
```suggestion
@Deprecated( forRemoval = true)
```

File: http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java [hunk 2]
```diff
@@ -23,7 +23,9 @@
  *
  * @author James Kleeh
  * @since 1.0
+ * @deprecated Deprecated without replacement. This class is unused and will be removed in 4.0.0.
```
Comment (sdelamo) [inline http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java:26]: Can you clarify in the javadoc what users should use instead?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java (hunk 1 above)
Comment (yawkat) [reply http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java:28]: the pr is to 3.9 so we can't use forRemoval i dont think?

File: http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java (hunk 2 above)
Comment (yawkat) [reply http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java:26]: there is no replacement, i dont know what this class was for and how it was supposed to be used. it was never referenced anywhere

File: http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java (hunk 1 above)
Comment (sdelamo) [reply http/src/main/java/io/micronaut/http/filter/FilterOrderProvider.java:28]: true, that is Java 9 only

**Review by sdelamo — APPROVED** [review]

**Pull request merged**